In [1]:
import pandas as pd
import pymysql
pymysql.install_as_MySQLdb()
import MySQLdb
import mysql.connector
import pymysql
import MySQLdb
from sqlalchemy import create_engine

In [2]:
import pandas as pd
import pymysql
pymysql.install_as_MySQLdb()
import MySQLdb
import mysql.connector
from sqlalchemy import create_engine
import numpy as np
from tqdm import trange
import time
import string
import re
import matplotlib.pyplot as plt
from bs4 import BeautifulSoup
import requests
import pymysql
pymysql.install_as_MySQLdb()
import pyodbc
import sqlalchemy
import MySQLdb
from sqlalchemy import create_engine
import mysql.connector
import pandas as pd
import xlrd
# from fuzzywuzzy import fuzz
import re
import wikipedia
from collections import Counter
from tqdm import tqdm
import time
import rapidfuzz
from fuzzywuzzy import process, fuzz

/home/jupyter/.local/lib/python3.6/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')


In [47]:
pd.set_option('display.max_columns', 100)

In [3]:
db = pymysql.connect(
                host = 'localhost',
                user = 'zshen',
                password = 'curly-straw-2275',
                database = 'colleges')  # create the connection

cursor = db.cursor()    # get the cursor

In [4]:
cursor.execute("show databases")
databases = cursor.fetchall()
databases

(('colleges',), ('information_schema',), ('performance_schema',))

In [5]:
cursor.execute("USE colleges")
databases = cursor.fetchall()

In [8]:
cursor.execute("show tables")
cursor.fetchall()
#tables

(('Campus_Price',),
 ('NCAA',),
 ('NICHE',),
 ('SC',),
 ('School_ID',),
 ('USNEWS',),
 ('Updated_income',),
 ('WG',),
 ('airports',),
 ('coalitionApp',),
 ('colleges.SC',),
 ('colleges.SC_test',),
 ('colleges.SC_test1',),
 ('combined_data_V4',),
 ('combined_data_V5',),
 ('commonApp',),
 ('db_log',),
 ('koppenClimate',),
 ('locationLocaleType',),
 ('locationRegion',),
 ('majors_desc',),
 ('manualTuition',),
 ('others',),
 ('payScale',),
 ('publicColleges',),
 ('religion',),
 ('schools',),
 ('schools_majors',),
 ('wikipedia',))

In [10]:
query1 = '''select * from schools'''
table = pd.read_sql(query1, db)
schools = table.copy()
schools

index  mykCollegeId                                           arcoName  \
0         0        1001.0                                  Auburn University   
1         1        1002.0                        Birmingham Southern College   
2         2        1003.0                          Concordia College Alabama   
3         3        1004.0                           CollegeAmerica-Flagstaff   
4         4        1005.0                                    Hendrix College   
...     ...           ...                                                ...   
7566   7566        9435.0                    Urban Barber College - San Jose   
7567   7567        9436.0              Suncoast Technical Education Center -   
7568   7568        9437.0  Western Maricopa Education Center - Southwest ...   
7569   7569        9438.0  Western Maricopa Education Center - Northeast ...   
7570   7570        9439.0  Pennsylvania State University-Penn State Shenango   

     ownershipType schoolDurationYears           schoolAddress         city  \
0           PUBLIC                FOUR             Quad Center       Auburn   
1          PRIVATE                FOUR    900 Arkadelphia Road   Birmingham   
2          PRIVATE                FOUR       1712 Broad Street        Selma   
3          PRIVATE                FOUR        399 S Malpais Ln    Flagstaff   
4          PRIVATE                FOUR  1600 Washington Avenue       Conway   
...            ...                 ...                     ...          ...   
7566    FOR_PROFIT       LESS_THAN_TWO                    None     San Jose   
7567        PUBLIC       LESS_THAN_TWO                    None  Brooksville   
7568        PUBLIC       LESS_THAN_TWO                    None      Buckeye   
7569        PUBLIC       LESS_THAN_TWO                    None      Phoenix   
7570        PUBLIC                FOUR                    None       Sharon   

     state    zip  latitude  longitude  isActive                    schoolURL  \
0       AL  36849   32.5994   -85.4883       1.0                  auburn.edu/   
1       AL  35204   33.5138   -86.8506       1.0                     bsc.edu/   
2       AL  36703   32.4256   -87.0191       0.0                         None   
3       AZ  86001   35.1945  -111.6580       0.0                         None   
4       AR  72032   35.1016   -92.4433       1.0                 hendrix.edu/   
...    ...    ...       ...        ...       ...                          ...   
7566    CA  95133   37.3729    37.3729       1.0      urbanbarbercollege.com/   
7567    FL  34613   28.5466    28.5466       1.0               sunteched.edu/   
7568    AZ  85326   33.3703    33.3703       1.0  west-mec.edu/findyourhappy/   
7569    AZ  85027   33.6819    33.6819       1.0  west-mec.edu/findyourhappy/   
7570    PA  16146   41.2316    41.2316       1.0            shenango.psu.edu/   

      inUSA  
0         1  
1         1  
2         1  
3         1  
4         1  
...     ...  
7566      1  
7567      1  
7568      1  
7569      1  
7570      1  

[7571 rows x 14 columns]

In [52]:
query1 = '''select * from School_ID'''
table = pd.read_sql(query1, db)
School_ID = table.copy()
School_ID

mykCollegeId    unitID       scid      opeId
0             1001  100858.0  2400026.0   100900.0
1             1002  100937.0  2400045.0   101200.0
2             1003  101073.0  3000146.0        NaN
3             1004  103945.0        NaN  3120300.0
4             1005  107080.0  2400118.0   109900.0
...            ...       ...        ...        ...
6870          8743       NaN        7.0        NaN
6871          8744       NaN        8.0        NaN
6872          8745       NaN        9.0        NaN
6873          8746       NaN       10.0        NaN
6874          8747       NaN  3000330.0        NaN

[6875 rows x 4 columns]

In [58]:
#School_ID.to_csv("School_ID_backup.csv", index = False)

In [81]:
query1 = '''select a.mykCollegeId,b.arcoName,b.city,b.state from School_ID a left join schools b on a.mykCollegeId = b.mykCollegeId '''
table = pd.read_sql(query1, db)
joined = table.copy()
joined

mykCollegeId                           arcoName         city state
0             1001                  Auburn University       Auburn    AL
1             1002        Birmingham Southern College   Birmingham    AL
2             1003          Concordia College Alabama        Selma    AL
3             1004           CollegeAmerica-Flagstaff    Flagstaff    AZ
4             1005                    Hendrix College       Conway    AR
...            ...                                ...          ...   ...
6870          8740      Institute of Buddhist Studies     Berkeley    CA
6871          8741                   Maitripa College     Portland    OR
6872          8742         Soka University of America  Aliso Viejo    CA
6873          8744  Won Institute of Graduate Studies     Glenside    PA
6874          8745           American Islamic College      Chicago    IL

[6875 rows x 4 columns]

In [82]:
def drop_pun(str):
    str = re.sub(r'[^\w\s]','',str)
    str = re.sub(' ','',str)
    str = re.sub('university','1',str)
    str = re.sub('college','2',str)
    str = re.sub('the','3',str)
    return str

def to_lower(item):
    return item.lower()

#
def stringCleaning(str):
    if str is None:
        return None
    try:
        str = to_lower(str)
        str = drop_pun(str)
        
        return str
    except:
        return None

stringCleaning('ab.`college')

'ab2'

In [7]:
Campus_price = pd.read_excel("CampusPride - LGBTQ Data-2021-09.xlsx")
Campus_price

CampusKey                                         CampusName  \
0          103                            Kansas State University   
1           36  Rutgers, The State University of New Jersey-Ne...   
2           50                                     Ithaca College   
3          485            Southern Illinois University Carbondale   
4           15               University of Maryland, College Park   
..         ...                                                ...   
449        383  City Colleges of Chicago-Richard J. Daley College   
450       2211                        Concordia University-Irvine   
451        380                        Washtenaw Community College   
452       7441  Vatel International Business School of Hospita...   
453       7437                              Berkeley City College   

              City State         Zip  StarRatingOverall  
0        Manhattan    KS       66506                5.0  
1    New Brunswick    NJ       08901                5.0  
2           Ithaca    NY       14850                5.0  
3       Carbondale    IL       62901                5.0  
4     College Park    MD       20742                5.0  
..             ...   ...         ...                ...  
449        Chicago    IL       60652                1.0  
450         Irvine    CA  92612-3203                1.0  
451      Ann Arbor    MI       48105                1.0  
452    Los Angeles    CA       90010                1.0  
453       Berkeley    CA       94704                1.0  

[454 rows x 6 columns]

In [84]:
num = 0
for i in joined["arcoName"]:
    for j in Campus_price["CampusName"]:
        if i == j:
            num += 1
num

344

In [86]:
import rapidfuzz
from fuzzywuzzy import process, fuzz

In [88]:
stringCleaning(Campus_price["CampusName"])

In [89]:
def DataClean(table, column1, column2, column3, newcolumn1, newcolumn2, newcolumn3):
    
    for i in range(len(table)):
        table.loc[i, newcolumn1] = stringCleaning(table[column1][i])
        table.loc[i, newcolumn2] = stringCleaning(table[column2][i])
        table.loc[i, newcolumn3] = stringCleaning(table[column3][i])

In [90]:
DataClean(joined, 'arcoName', 'city', 'state', 'Tb1_arcoName', 'Tb1_city', 'Tb1_state')
joined

mykCollegeId                           arcoName         city state  \
0             1001                  Auburn University       Auburn    AL   
1             1002        Birmingham Southern College   Birmingham    AL   
2             1003          Concordia College Alabama        Selma    AL   
3             1004           CollegeAmerica-Flagstaff    Flagstaff    AZ   
4             1005                    Hendrix College       Conway    AR   
...            ...                                ...          ...   ...   
6870          8740      Institute of Buddhist Studies     Berkeley    CA   
6871          8741                   Maitripa College     Portland    OR   
6872          8742         Soka University of America  Aliso Viejo    CA   
6873          8744  Won Institute of Graduate Studies     Glenside    PA   
6874          8745           American Islamic College      Chicago    IL   

                       Tb1_arcoName    Tb1_city Tb1_state  
0                           auburn1      auburn        al  
1                 birminghamsou3rn2  birmingham        al  
2                 concordia2alabama       selma        al  
3                 2americaflagstaff   flagstaff        az  
4                          hendrix2      conway        ar  
...                             ...         ...       ...  
6870     instituteofbuddhiststudies    berkeley        ca  
6871                      maitripa2    portland        or  
6872                 soka1ofamerica  alisoviejo        ca  
6873  woninstituteofgraduatestudies    glenside        pa  
6874               americanislamic2     chicago        il  

[6875 rows x 7 columns]

In [91]:
DataClean(Campus_price, 'CampusName', 'City', 'State', 'Tb2_arcoName', 'Tb2_city', 'Tb2_state')
Campus_price

CampusKey                                         CampusName  \
0          103                            Kansas State University   
1           36  Rutgers, The State University of New Jersey-Ne...   
2           50                                     Ithaca College   
3          485            Southern Illinois University Carbondale   
4           15               University of Maryland, College Park   
..         ...                                                ...   
449        383  City Colleges of Chicago-Richard J. Daley College   
450       2211                        Concordia University-Irvine   
451        380                        Washtenaw Community College   
452       7441  Vatel International Business School of Hospita...   
453       7437                              Berkeley City College   

              City State         Zip  StarRatingOverall  \
0        Manhattan    KS       66506                5.0   
1    New Brunswick    NJ       08901                5.0   
2           Ithaca    NY       14850                5.0   
3       Carbondale    IL       62901                5.0   
4     College Park    MD       20742                5.0   
..             ...   ...         ...                ...   
449        Chicago    IL       60652                1.0   
450         Irvine    CA  92612-3203                1.0   
451      Ann Arbor    MI       48105                1.0   
452    Los Angeles    CA       90010                1.0   
453       Berkeley    CA       94704                1.0   

                                          Tb2_arcoName      Tb2_city Tb2_state  
0                                         kansasstate1     manhattan        ks  
1                rutgers3state1ofnewjerseynewbrunswick  newbrunswick        nj  
2                                              ithaca2        ithaca        ny  
3                            sou3rnillinois1carbondale    carbondale        il  
4                                     1ofmaryland2park         2park        md  
..                                                 ...           ...       ...  
449                      city2sofchicagorichardjdaley2       chicago        il  
450                                   concordia1irvine        irvine        ca  
451                                washtenawcommunity2      annarbor        mi  
452  vatelinternationalbusinessschoolofhospitalityt...    losangeles        ca  
453                                      berkeleycity2      berkeley        ca  

[454 rows x 9 columns]

In [24]:
def matchHelper(tg1, tg2, tg3, cl1, cl2, cl3, table1, model='model2'):
    max_score = -1
    max_name = ''
    max_pk = ''
#     count = 0
    for i in range(len(table1)):
#         if count == 3:
#             break
#         count = count + 1
        pk = table1['mykCollegeId'][i]
        if model == 'model1':
            score1 = fuzz.token_set_ratio(tg1, table1[cl1][i]) 
            score = score1
        elif model == 'model2':
            score1 = fuzz.token_set_ratio(tg1, table1[cl1][i])
            score2 = fuzz.partial_ratio(tg2, table1[cl2][i]) 
            score3 = fuzz.partial_ratio(tg3, table1[cl3][i])
            if (score2 > 75 & score3 > 75):
                score = score1 * 50 / 100 + score2 * 25 / 100 + score3 * 25 / 100
            else:
                score = score1 * 20 / 100 + score2 * 40 / 100 + score3 * 40 / 100
        else:
            print('Wrong model!')
            return ('',0,'')
            
        #school name
#         score1 = fuzz.token_set_ratio(tg1, table1[cl1][i])
      
#         #city
#         score2 = fuzz.partial_ratio(tg2, table1[cl2][i])
        
# #         #state
#         score3 = fuzz.partial_ratio(tg3, table1[cl3][i])
        
# #         #url
#         score4 = fuzz.token_sort_ratio (tg4, table1[cl4][i])
#         print(str(score1) +'--------'+ tg1 +'-----' + table1[cl1][i])
#         print(str(score2) +'--------'+ tg2 +'-----' + table1[cl2][i])
#         print(str(score3) +'--------'+ tg3 +'-----' + table1[cl3][i])
#         print(str(score4) +'--------'+ tg4 +'-----' + table1[cl4][i])
#         print("-------------------------------------------------------------")

        
        
#         score = (score1 + score2 + score3 + score4) / 4
#         print('score: ' + str(score) + '/' + 'name_score1: ' + str(score1) + '/' + 'city_score2: ' + str(score2) + '/'
#              + 'state_score3: ' + str(score3) + '/')
        if (score > 75) & (score > max_score):
            max_name = table1[cl1][i]
            max_score = score
            max_pk = pk
            
    return (max_name, max_score, max_pk)

In [25]:
# table1  old table in myklovr
# table2  new table waiting to match
# name_dict = {}
# fail_dict = {}
def FuzzyMatch(table1, table2, tb1_column1, tb1_column2, tb1_column3, tb2_column1, tb2_column2, tb2_column3, model):
    names = []
    res = []
    fails = []
#     count = 0
    for i in range(len(table2)):
#         if count == 2:
#             break
#         count = count + 1
        match = matchHelper(table2[tb2_column1][i], table2[tb2_column2][i], table2[tb2_column3][i], tb1_column1, tb1_column2, tb1_column3, table1, model)
#         print(match)
        if match[1] >= 80:
            name = (str(match[2]), str(match[0]) + ' | ' + str(table2[tb2_column1][i]) + ' | ' + str(match[1]))
            print(name)
            names.append(name)
            temp = (str(match[2]), str(table2[tb2_column1][i]))
            print(temp)
            res.append(temp)
        else:
            fail = (str(match[2]), str(match[0]) + ' | ' + str(table2[tb2_column1][i]) + ' | ' + str(match[1]))
#             print(fail)
            fails.append(fail)
    name_dict = dict(names)
    res_dict = dict(res)
    fails_dict = dict(fails)
    return (name_dict, res_dict)

In [26]:
res = FuzzyMatch(joined, Campus_price, 'Tb1_arcoName', 'Tb1_city', 'Tb1_state', 'Tb2_arcoName', 'Tb2_city', 'Tb2_state', 'model2')



('8057', 'kansasstate1 | kansasstate1 | 100.0')
('8057', 'kansasstate1')
('8735', 'rutgers3state1ofnewjerseyhonors2 | rutgers3state1ofnewjerseynewbrunswick | 96.2')
('8735', 'rutgers3state1ofnewjerseynewbrunswick')
('1599', 'ithaca2 | ithaca2 | 100.0')
('1599', 'ithaca2')
('7845', 'sou3rnillinois1carbondale | sou3rnillinois1carbondale | 100.0')
('7845', 'sou3rnillinois1carbondale')
('8282', '1ofmaryland2park | 1ofmaryland2park | 100.0')
('8282', '1ofmaryland2park')
('3141', '1ofwashingtonseattlecampus | 1ofwashington | 93.4')
('3141', '1ofwashington')
('2243', '1oforegon | 1oforegon | 100.0')
('2243', '1oforegon')
('2252', 'portlandstate1 | portlandstate1 | 100.0')
('2252', 'portlandstate1')
('7334', '1ofcoloradoboulder | 1ofcoloradoatboulder | 99.0')
('7334', '1ofcoloradoatboulder')
('8375', '1ofmassachusettsamherst | 1ofmassachusettsamherst | 100.0')
('8375', '1ofmassachusettsamherst')
('7897', 'indiana1bloomington | indiana1bloomington | 100.0')
('7897', 'indiana1bloomington')
('324

('1388', 'rowan1 | rowan1 | 100.0')
('1388', 'rowan1')
('3252', '1ofwisconsinoshkosh | 1ofwisconsinoshkosh | 100.0')
('3252', '1ofwisconsinoshkosh')
('2500', 'swarthmore2 | swarthmore2 | 96.0')
('2500', 'swarthmore2')
('7361', 'metropolitanstate1ofdenver | metropolitanstate1ofdenver | 100.0')
('7361', 'metropolitanstate1ofdenver')
('3084', 'centralwashington1 | centralwashington1 | 100.0')
('3084', 'centralwashington1')
('2322', 'dickinson2 | dickinson2 | 100.0')
('2322', 'dickinson2')
('3063', 'virginiacommonwealth1 | virginiacommonwealth1 | 100.0')
('3063', 'virginiacommonwealth1')
('6894', 'nor3rnarizona1 | nor3rnarizona1 | 100.0')
('6894', 'nor3rnarizona1')
('7446', 'american1 | american1 | 100.0')
('7446', 'american1')
('3044', '1ofrichmond | 1ofrichmond | 100.0')
('3044', '1ofrichmond')
('3239', '1ofwisconsin2s | 1ofwisconsin | 98.4')
('3239', '1ofwisconsin')
('2806', '1ofhouston | 1ofhouston | 100.0')
('2806', '1ofhouston')
('1245', 'washington1instlouis | washington1instlouis |

('8299', 'towson1 | towson1 | 100.0')
('8299', 'towson1')
('2383', 'lebanonvalley2 | lebanonvalley2 | 100.0')
('2383', 'lebanonvalley2')
('8267', 'goucher2 | goucher2 | 100.0')
('8267', 'goucher2')
('8558', 'saintcloudstate1 | stcloudstate1 | 92.4')
('8558', 'stcloudstate1')
('7234', '1ofredlands | 1ofredlands | 100.0')
('7234', '1ofredlands')
('7451', 'georgewashington1 | 3georgewashington1 | 99.4')
('7451', '3georgewashington1')
('7416', '1ofnewhaven | 1ofnewhaven | 100.0')
('7416', '1ofnewhaven')
('2254', 'reed2 | reed2 | 100.0')
('2254', 'reed2')
('1005', 'hendrix2 | hendrix2 | 100.0')
('1005', 'hendrix2')
('7352', '1ofdenver | 1ofcoloradodenver | 93.8')
('7352', '1ofcoloradodenver')
('8426', 'worcesterpolytechnicinstitute | worcesterpolytechnicinstitute | 100.0')
('8426', 'worcesterpolytechnicinstitute')
('1436', 'stockton1 | stockton1 | 100.0')
('1436', 'stockton1')
('7012', 'californiapolytechnicstate1sanluisobispo | californiapolytechnicstate1sanluisobispo | 100.0')
('7012', 'c

('1344', 'plymouthstate1 | plymouth1 | 95.6')
('1344', 'plymouth1')
('3260', '1ofwisconsinriverfalls | 1ofwisconsinriverfalls | 100.0')
('3260', '1ofwisconsinriverfalls')
('3259', '1ofwisconsinplatteville | 1ofwisconsinplatteville | 100.0')
('3259', '1ofwisconsinplatteville')
('2364', 'indiana1ofpennsylvaniamaincampus | indiana1ofpennsylvania | 96.2')
('2364', 'indiana1ofpennsylvania')
('8102', 'easternkentucky1 | easternkentucky1 | 100.0')
('8102', 'easternkentucky1')
('1424', 'rutgers1camden | rutgers3state1camdencampus | 94.0')
('1424', 'rutgers3state1camdencampus')
('2009', 'heidelberg1 | heidelberg1 | 100.0')
('2009', 'heidelberg1')
('1872', 'queens1ofcharlotte | queens1ofcharlotte | 100.0')
('1872', 'queens1ofcharlotte')
('2269', 'westernoregon1 | westernoregon1 | 100.0')
('2269', 'westernoregon1')
('1345', 'nhticoncordscommunity2 | nhticoncordscommunity2 | 100.0')
('1345', 'nhticoncordscommunity2')
('1539', 'cooperunionfor3advancementofscienceandart | 3cooperunion | 88.4')
('153

('2331', 'elizabethtown2 | elizabethtown2 | 100.0')
('2331', 'elizabethtown2')
('7270', 'santarosajunior2 | santarosajunior2 | 100.0')
('7270', 'santarosajunior2')
('8460', 'henryford2 | henryfordcommunity2 | 93.8')
('8460', 'henryfordcommunity2')
('6933', '1ofcentralarkansas | 1ofcentralarkansas | 100.0')
('6933', '1ofcentralarkansas')
('3126', 'southseattle2 | southseattle2 | 100.0')
('3126', 'southseattle2')
('7174', '2ofmarin | 2ofmarin | 100.0')
('7174', '2ofmarin')
('1215', 'northwestmissouristate1 | northwestmissouristate1 | 100.0')
('1215', 'northwestmissouristate1')
('2955', 'utahstate1 | utahstate1 | 100.0')
('2955', 'utahstate1')
('1839', 'marshill1 | marshill1 | 100.0')
('1839', 'marshill1')
('7391', '1ofbridgeport | 1ofbridgeport | 100.0')
('7391', '1ofbridgeport')
('1799', 'catawba2 | catawba2 | 100.0')
('1799', 'catawba2')
('8535', 'metropolitanstate1 | metropolitanstate1 | 100.0')
('8535', 'metropolitanstate1')
('1300', '1ofnebraskamedicalcenter | 1ofnebraskamedicalcent

In [34]:
joined.loc[joined["mykCollegeId"] == 8057,:]

mykCollegeId                 arcoName       city state  Tb1_arcoName  \
6559          8057  Kansas State University  Manhattan    KS  kansasstate1   

       Tb1_city Tb1_state  
6559  manhattan        ks

In [58]:
import pandas as pd
 
my_dict = data
my_df = pd.DataFrame(my_dict,index=["Tb2_arcoName"]).T
my_df['matched_myid'] = my_df.index
my_df

Tb2_arcoName matched_myid
8057                                       kansasstate1         8057
8735              rutgers3state1ofnewjerseynewbrunswick         8735
1599                                            ithaca2         1599
7845                          sou3rnillinois1carbondale         7845
8282                                   1ofmaryland2park         8282
...                                                 ...          ...
7717                      city2sofchicagorichardjdaley2         7717
7060                                   concordia1irvine         7060
8500                                washtenawcommunity2         8500
5657  vatelinternationalbusinessschoolofhospitalityt...         5657
7309                                      berkeleycity2         7309

[445 rows x 2 columns]

In [54]:
Campus_price["Tb2_arcoName"]

0                                           kansasstate1
1                  rutgers3state1ofnewjerseynewbrunswick
2                                                ithaca2
3                              sou3rnillinois1carbondale
4                                       1ofmaryland2park
                             ...                        
449                        city2sofchicagorichardjdaley2
450                                     concordia1irvine
451                                  washtenawcommunity2
452    vatelinternationalbusinessschoolofhospitalityt...
453                                        berkeleycity2
Name: Tb2_arcoName, Length: 454, dtype: object

In [62]:
get_myid = pd.merge(my_df, Campus_price, on='Tb2_arcoName',how='right')

In [116]:
get_myid.iloc[445:454,:]

Tb2_arcoName matched_myid  CampusKey  \
445                  virginiawesleyan1         3070       2262   
446                        macalester2         8529        247   
447                        alfredstate         1484        503   
448                  1ofcoloradodenver         7331        307   
449                   3american1ofrome          NaN       8491   
450        1ofnorthcarolinaatcharlotte         7331       5702   
451    cuyahogacommunity2easterncampus          NaN       8470   
452  cuyahogacommunity2westshorecampus          NaN       8471   
453                        capebreton1          NaN       8523   

                                      CampusName            City State  \
445                 Virginia Wesleyan University  Virginia Beach    VA   
446                           Macalester College      Saint Paul    MN   
447                                 Alfred State          Alfred    NY   
448                University of Colorado Denver          Denver    CO   
449              The American University of Rome            Rome    DC   
450    University of North Carolina at Charlotte       Charlotte    NC   
451    Cuyahoga Community College Eastern Campus  Highland Hills    OH   
452  Cuyahoga Community College Westshore Campus        Westlake    OH   
453                       Cape Breton University          Sydney    ME   

            Zip  StarRatingOverall       Tb2_city Tb2_state  
445       23455                5.0  virginiabeach        va  
446       55105                5.0      saintpaul        mn  
447       14802                4.0         alfred        ny  
448       80204                3.5         denver        co  
449       00153                3.5           rome        dc  
450  28223-0001                3.0      charlotte        nc  
451       44122                3.0  highlandhills        oh  
452       44145                3.0       westlake        oh  
453     B1P 6L2                2.0         sydney        me

In [107]:
joined.loc[joined["Tb1_city"] == "sydney",:]

Empty DataFrame
Columns: [mykCollegeId, arcoName, city, state, Tb1_arcoName, Tb1_city, Tb1_state]
Index: []

3070	Virginia Wesleyan University	Virginia Beach | virginiawesleyan1

8529	Macalester College	Saint Paul    | macalester2

1484	Alfred University	Alfred	NY	alfred1	alfred	ny | alfredstate

7331	University of Colorado Denver/Anschutz Medical...	Denver	CO | 1ofcoloradodenver

3american1ofrome	NaN	   8491	The American University of Rome	

1858	University of North Carolina at Charlotte	Charlotte	NC	1ofnorthcarolinaatcharlotte

nan

nan

nan







In [ ]:
get_myid.rename(columns = {'matched_myid':'mykCollegeId'}, inplace = True)
get_myid_2 = get_myid[["mykCollegeId","CampusKey"]]
get_myid_2 = get_myid_2.dropna(subset=['mykCollegeId'])

In [145]:
get_myid_2

mykCollegeId  CampusKey
0           8057        103
1           8735         36
2           1599         50
3           7845        485
4           8282         15
..           ...        ...
445         3070       2262
446         8529        247
447         1484        503
448         7331        307
450         7331       5702

[450 rows x 2 columns]

In [147]:
get_myid_2['mykCollegeId'] = get_myid_2['mykCollegeId'].astype(int)

/home/jupyter/miniconda3/envs/py36/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


# Join to the school_ID table

In [148]:
School_ID

mykCollegeId    unitID       scid      opeId
0             1001  100858.0  2400026.0   100900.0
1             1002  100937.0  2400045.0   101200.0
2             1003  101073.0  3000146.0        NaN
3             1004  103945.0        NaN  3120300.0
4             1005  107080.0  2400118.0   109900.0
...            ...       ...        ...        ...
6870          8743       NaN        7.0        NaN
6871          8744       NaN        8.0        NaN
6872          8745       NaN        9.0        NaN
6873          8746       NaN       10.0        NaN
6874          8747       NaN  3000330.0        NaN

[6875 rows x 4 columns]

In [141]:
type(School_ID.loc[School_ID["mykCollegeId"] == 8057,"mykCollegeId"])

pandas.core.series.Series

In [149]:
Matched_schoolsID = pd.merge(School_ID, get_myid_2, on='mykCollegeId',how='left')

In [150]:
Matched_schoolsID

mykCollegeId    unitID       scid      opeId  CampusKey
0             1001  100858.0  2400026.0   100900.0      644.0
1             1002  100937.0  2400045.0   101200.0        NaN
2             1003  101073.0  3000146.0        NaN        NaN
3             1004  103945.0        NaN  3120300.0        NaN
4             1005  107080.0  2400118.0   109900.0      252.0
...            ...       ...        ...        ...        ...
6874          8743       NaN        7.0        NaN        NaN
6875          8744       NaN        8.0        NaN        NaN
6876          8745       NaN        9.0        NaN        NaN
6877          8746       NaN       10.0        NaN        NaN
6878          8747       NaN  3000330.0        NaN        NaN

[6879 rows x 5 columns]

In [153]:
Matched_schoolsID["campuskey"].notnull().sum()

450

In [2]:
import pandas as pd
Matched_schoolsID = pd.read_csv("Updated_schoolsID-campuskey.csv")
#Matched_schoolsID.rename(columns = {'campuskey':'CampusKey'}, inplace = True)

In [3]:
Matched_schoolsID

mykCollegeId    unitID       scid      opeId  CampusKey
0             1001  100858.0  2400026.0   100900.0      644.0
1             1002  100937.0  2400045.0   101200.0        NaN
2             1003  101073.0  3000146.0        NaN        NaN
3             1004  103945.0        NaN  3120300.0        NaN
4             1005  107080.0  2400118.0   109900.0      252.0
...            ...       ...        ...        ...        ...
6874          8743       NaN        7.0        NaN        NaN
6875          8744       NaN        8.0        NaN        NaN
6876          8745       NaN        9.0        NaN        NaN
6877          8746       NaN       10.0        NaN        NaN
6878          8747       NaN  3000330.0        NaN        NaN

[6879 rows x 5 columns]

In [8]:
Campus_price

CampusKey                                         CampusName  \
0          103                            Kansas State University   
1           36  Rutgers, The State University of New Jersey-Ne...   
2           50                                     Ithaca College   
3          485            Southern Illinois University Carbondale   
4           15               University of Maryland, College Park   
..         ...                                                ...   
449        383  City Colleges of Chicago-Richard J. Daley College   
450       2211                        Concordia University-Irvine   
451        380                        Washtenaw Community College   
452       7441  Vatel International Business School of Hospita...   
453       7437                              Berkeley City College   

              City State         Zip  StarRatingOverall  
0        Manhattan    KS       66506                5.0  
1    New Brunswick    NJ       08901                5.0  
2           Ithaca    NY       14850                5.0  
3       Carbondale    IL       62901                5.0  
4     College Park    MD       20742                5.0  
..             ...   ...         ...                ...  
449        Chicago    IL       60652                1.0  
450         Irvine    CA  92612-3203                1.0  
451      Ann Arbor    MI       48105                1.0  
452    Los Angeles    CA       90010                1.0  
453       Berkeley    CA       94704                1.0  

[454 rows x 6 columns]

In [9]:
CP = pd.merge(Campus_price, Matched_schoolsID, on = "CampusKey", how = "left")
CP_for_sql = CP[["mykCollegeId","CampusKey","StarRatingOverall"]]
CP_for_sql

mykCollegeId  CampusKey  StarRatingOverall
0          8057.0        103                5.0
1          8735.0         36                5.0
2          1599.0         50                5.0
3          7845.0        485                5.0
4          8282.0         15                5.0
..            ...        ...                ...
449        7717.0        383                1.0
450        7060.0       2211                1.0
451        8500.0        380                1.0
452        5657.0       7441                1.0
453        7309.0       7437                1.0

[454 rows x 3 columns]

In [10]:
#CP_for_sql.to_csv("CP_for_sql.csv",index=False)

In [4]:
CP_for_sql = pd.read_csv("CP_for_sql.csv")
CP_for_sql

mykCollegeId  CampusKey  StarRatingOverall
0          8057.0        103                5.0
1          8735.0         36                5.0
2          1599.0         50                5.0
3          7845.0        485                5.0
4          8282.0         15                5.0
..            ...        ...                ...
449        7717.0        383                1.0
450        7060.0       2211                1.0
451        8500.0        380                1.0
452        5657.0       7441                1.0
453        7309.0       7437                1.0

[454 rows x 3 columns]